src/core/Akka/Actor/FSM.cs

```diff
@@ -1100,20 +1100,20 @@ private static StateFunction OrElse(StateFunction original, StateFunction fallba
         /// <inheritdoc/>
         protected override bool Receive(object message)
         {
-            var timeoutMarker = message as TimeoutMarker;
-            if (timeoutMarker != null)
+            switch (message)
             {
-                if (_generation == timeoutMarker.Generation)
+                case TimeoutMarker timeoutMarker:
                 {
-                    ProcessMsg(StateTimeout.Instance, "state timeout");
+                    if (_generation == timeoutMarker.Generation)
+                    {
+                        ProcessMsg(StateTimeout.Instance, "state timeout");
+                    }
+                    return true;
                 }
-                return true;
-            }
-
-            if (message is Timer timer)
-            {
-                if (ReferenceEquals(timer.Owner, this) && _timers.TryGetValue(timer.Name, out var oldTimer) && oldTimer.Generation == timer.Generation)
+                case Timer timer:
                 {
+                    if (!ReferenceEquals(timer.Owner, this) || !_timers.TryGetValue(timer.Name, out var oldTimer) ||
+                        oldTimer.Generation != timer.Generation) return true;
```

Comment:
Can we move this `return true` statement to the new line, or reformat thing somehow to see it more clearly here?

Reply:
Yeah I'll take a look at cleaning that up - the original code was pretty messy too

Reply:
So now this is going to be the only change I would like to suggest for this PR

Reply:
Resolved this, I believe